source/WindowScroller/utils/dimensions.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Gets the vertical scroll amount of the element, accounting for IE compatibility
+ * and API differences between `window` and other DOM elements.
+ */
+export function getVerticalScroll (element) {
+  return element === window
+    ? (('scrollY' in window) ? window.scrollY : document.documentElement.scrollTop)
+    : element.scrollTop
+}
+
+/**
+ * Gets the height of the element, accounting for API differences between
+ * `window` and other DOM elements.
+ */
+export function getHeight (element) {
+  return element === window
+    ? window.innerHeight
+    : element.getBoundingClientRect().height
+}
+
+/**
+ * Gets the vertical position of an element within its scroll container.
+ * Elements that have been “scrolled past” return negative values.
+ * Handles edge-case where a user is navigating back (history) from an already-scrolled page.
+ * In this case the body’s top position will be a negative number and this element’s top will be increased (by that amount).
+ */
+export function getPositionFromTop (element, container) {
```

Comment:
I was having trouble wrapping my mind around this (previously [here](https://github.com/bvaughn/react-virtualized/pull/481/files#diff-668ee30b9b9d3323692d842bec7a5f55L106)), so extra care in making sure this satisfies the original intention is appreciated.